### PR TITLE
[Docs] Added/Updated Lib builder docs

### PR DESCRIPTION
**Pull request by pedrominatel**
## Summary

Update on the library builder procedures.

## Impact

None

## Related links

https://github.com/espressif/arduino-esp32/projects/5#card-63388323

### CODE REVIEW

File: docs/source/lib_builder.rst [hunk 1]
```diff
@@ -2,37 +2,173 @@
...
 
-- Step 6 - Build::
+Specify additional configs to be applied. ex. 'qio 80m' to compile for QIO Flash@80MHz. Requires -b
 
-    idf.py build
+.. code-block:: bash
 
-The script automates the process of building `Arduino as an ESP-IDF component <https://github.com/espressif/arduino-esp32/blob/master/docs/esp-idf_component.md>`_.
-Once it is complete, you can cherry pick the needed libraries from ``out/tools/sdk/lib``, or run ``tools/copy-to-arduino.sh`` to copy the entire built system.
-``tools/config.sh`` contains a number of variables that control the process, particularly the ``$IDF_BRANCH`` variable.  You can adjust this to try building against newer versions, but there are absolutely no guarantees that any components will work or even successfully compile against a newer IDF.
```
Comment (VojtechBartoska) [inline docs/source/lib_builder.rst:38]: This part is not "valid" anymore? Otherwise looks good.

**Review by VojtechBartoska — APPROVED** [review]

File: docs/source/lib_builder.rst (hunk 1 above)
Comment (me-no-dev) [reply docs/source/lib_builder.rst:38]: not valid anymore, though it will still work, there is a better way :)

File: docs/source/lib_builder.rst [hunk 2]
```diff
@@ -2,37 +2,173 @@
...
+* esp32s2
+* esp32c3
+* esp32s3
 
-    ./tools/update-components.sh`
+Set Build Type
+^^^^^^^^^^^^^^
 
-- Step 4 - Run ``install-esp-idf`` installation script (if you already have an ``$IDF_PATH`` defined, it will use your local copy of the repository)::
+Set the build type. ex. 'build' to build the project and prepare for uploading to a board
```
Comment (me-no-dev) [inline docs/source/lib_builder.rst:161]: depends on `-t`

File: docs/source/lib_builder.rst [hunk 3]
```diff
@@ -2,37 +2,173 @@
...
-- Step 6 - Build::
+Specify additional configs to be applied. ex. 'qio 80m' to compile for QIO Flash@80MHz. Requires -b
 
-    idf.py build
+.. code-block:: bash
 
-The script automates the process of building `Arduino as an ESP-IDF component <https://github.com/espressif/arduino-esp32/blob/master/docs/esp-idf_component.md>`_.
-Once it is complete, you can cherry pick the needed libraries from ``out/tools/sdk/lib``, or run ``tools/copy-to-arduino.sh`` to copy the entire built system.
-``tools/config.sh`` contains a number of variables that control the process, particularly the ``$IDF_BRANCH`` variable.  You can adjust this to try building against newer versions, but there are absolutely no guarantees that any components will work or even successfully compile against a newer IDF.
+    ./build.sh -b
```
Comment (me-no-dev) [inline docs/source/lib_builder.rst:174]: `./build.sh -t esp32 -b idf_libs qio 80m`

File: docs/source/lib_builder.rst [hunk 4]
```diff
@@ -2,37 +2,173 @@
...
+^^^^^^^^^^^^^^
 
-- Step 4 - Run ``install-esp-idf`` installation script (if you already have an ``$IDF_PATH`` defined, it will use your local copy of the repository)::
+Set the build type. ex. 'build' to build the project and prepare for uploading to a board
 
-    ./tools/install-esp-idf.sh
+.. code-block:: bash
 
-- Step 5 - Copy the configuration (recommended) or directly edit sdkconfig using ``idf.py menuconfig``::
+    ./build.sh -b <build|menuconfig|idf_libs|copy_bootloader|mem_variant>
```
Comment (me-no-dev) [inline docs/source/lib_builder.rst:165]: `./build.sh -t esp32 -b <build|menuconfig|idf_libs|copy_bootloader|mem_variant>`

File: docs/source/lib_builder.rst (hunk 2 above)
Comment (pedrominatel) [reply docs/source/lib_builder.rst]: PTAL

File: docs/source/lib_builder.rst (hunk 3 above)
Comment (pedrominatel) [reply docs/source/lib_builder.rst]: PTAL

**Comment by me-no-dev** [comment]
merged into S3 branch since it depends on it :)

**Pull request merged**